lib/rubycritic/generators/xml_report.rb

```diff
@@ -0,0 +1,24 @@
+require 'rubycritic/generators/xml/simple'
+
+module Rubycritic
+  module Generator
+    class XmlReport
+      def initialize(analysed_modules)
+        @analysed_modules = analysed_modules
+      end
+
+      def generate_report
+        report = generator.render
+
+        File.write "#{ Config.root }/report.xml", report
+        print report
+      end
+
+      private
+
+      def generator
+        Xml::Simple.new(@analysed_modules)
```

Comment:
I question this design even though it might be idiomatic for our current codebase. You are adding a hardcoded indirection from "base" to the "report" that offers no advantages but has the disadvantage of being less readable and confusing. If xml reporting was part of our domain i could see how we might have more than 1 strategy to generate xml reports but in its current state this reeks a bit of premature optimization.

I'd rather dissolve "base" into "xml_report" and keep it simple.